packages/block-library/src/comment-template/index.php

```diff
@@ -5,6 +5,84 @@
  * @package WordPress
  */
 
+/**
+ * Convert the flat array of comments to a nested array
+ * containing comments with a `comment_children` property.
+ *
+ * @param WP_Comment[] $comments  - Array of comments.
+ * @return array
+ */
+function convert_to_tree( $comments ) {
+	$table = array();
+
+	// If there are no comments, we can return early an empty array.
+	if ( ! isset( $comments ) || empty( $comments ) ) {
+		return array();
+	}
+
+	// First we create an associative array of the form:
+	// [commentId] => [...comment, comment_children: [] ].
+	foreach ( $comments as $comment ) {
+		$comment_with_children                     = (array) $comment;
+		$comment_with_children['comment_children'] = array();
+		$table[ $comment->comment_ID ]             = $comment_with_children;
+	}
+
+	$tree = array();
+
+	foreach ( $comments as $comment ) {
+		if ( ! empty( $comment->comment_parent ) ) {
+			// If the comment has a "parent", then find that parent in the table that
+			// we have created above and push the current comment to the array of its
+			// children.
+			$table[ $comment->comment_parent ]['comment_children'][] = $table[ $comment->comment_ID ];
+		} else {
+			// Otherwise, if the comment has no parent (also works if parent is 0/"0")
+			// that means that it's a top-level comment. So we can find it in the table
+			// and push it to the final tree.
+			//
+			// We have to push by reference using `=&` because this top-level comment might
+			// still have other children. This way, if we add another comment to the
+			// children of the current comment (in the `if` branch), it will be
+			// updated in the final tree.
+			$tree[] =& $table[ $comment->comment_ID ];
+		}
+	}
+
+	return $tree;
+}
+
+/**
+ * Function that recursively renders a list of nested comments.
+ *
+ * @param WP_Comment[] $comments    The array of comments.
+ * @param WP_Block     $block           Block instance.
+ * @return string
+ */
+function render_comments( $comments, $block ) {
```

Comment:
Nitpick: this file will eventually land in WordPress Core. Let's use less general name that is tied to the block name. You can check other blocks like the Navigation block where all PHP functions are prefixed with `block_core_navigation_`. The same comment applies to the `convert_to_tree` function.

Reply:
Done in e83166a